cpp/ql/src/semmle/code/cpp/Declaration.qll

```diff
@@ -281,8 +295,19 @@ abstract class DeclarationEntry extends Locatable {
  * A declaration that can potentially have more C++ access rights than its
  * enclosing element. This comprises `Class` (they have access to their own
  * private members) along with other `UserType`s and `Function` (they can be
- * the target of `friend` declarations).
- *
+ * the target of `friend` declarations).  For example `MyClass` and
+ * `myFunction` in the following code:
+ * ```
+ * class MyClass
+ * {
+ * public:
+ *   ...
+ * };
+ * 
+ * void myFunction() {
+ *   // ...
+ * }
+ * ```
```

Comment:
Is it worthwhile to have this exposition for an `abstract` class?  The `DirectAccessHolder` seems to be a more logical choice (and there is a QLDoc there already).

Reply:
I'm not sure why `AccessHolder` is declared `abstract`, because `DirectAccessHolder` doesn't actually extend it - it extends `Element` and is a private class that appears to exist to aid in the implementation of `AccessHolder`.

Reply:
It's `abstract` because its character is defined by the classes that extend it, such as `UserType`. The `DirectAccessHolder` class is private.

Reply:
So it does (sorry, my search didn't show that up).  Then yes, there's no need for an example on this abstract class, but I see no need to delete it either unless we think it's confusing or needs a lot more work?

Reply:
I did some more digging, and I think you (@geoffw0) were right about placing the QLDoc above `AccessHolder`.  `AccessHolder` is extended by `Function` and `UserType`, and it would not make sense to move the exposition there. 

I do not understand the interplay between `AccessHolder` and `DirectAccessHolder`, and so would personally be inclined to leave the QLDoc for the latter unchanged.